packages/cosmic-swingset/src/helpers/sync-xfer.js

```diff
@@ -0,0 +1,160 @@
+// @ts-check
+
+import { Far } from '@endo/far';
+
+// FIXME: for debugging the buffer splitting/rejoining:
+// const DEFAULT_BUFFER_SIZE = 12;
+const DEFAULT_BUFFER_SIZE = 64 * 1024;
+export { DEFAULT_BUFFER_SIZE };
+
+// const TRANSFER_UNSPECIFIED = 0;
+const TRANSFER_CLIENT_READY = 1;
+const TRANSFER_READY_TO_READ = 2;
+const TRANSFER_READ_PARTIAL = 3;
+const TRANSFER_READ_DONE = 4;
```

Comment:
_question_

Is there documentation I can read on how we transition between these states?

Reply:
> Is there documentation I can read on how we transition between these states?

I second this, and would also propose some renaming:
```suggestion
/**
 * TransferClient is waiting, either for permission to send a request elsewhere
 * (i.e., not to the TransferServer) that will result in TransferServer
 * receiving the response, or for permission to read a chunk of response data
 * from the TransferServer.
 */
const TRANSFER_CLIENT_WAITING = 1;
/**
 * TransferServer is ready to receive response data; TransferClient should send
 * its request.
 */
const TRANSFER_SERVER_READY = 2;
/**
 * TransferServer has written a non-final chunk of response data for
 * TransferClient to read and will sleep until TRANSFER_CLIENT_WAITING.
 */
const TRANSFER_DATA_AVAILABLE = 3;
/**
 * TransferServer has written a final chunk of response data for TransferClient
 * to read and then terminated itself.
 */
const TRANSFER_DATA_DONE = 4;
```
TRANSFER_DATA_DONE is also the de facto initial state and could have value 0.